### ISSUE TITLE
RSpec is not taking the baseurl with api namespaces into account

### ISSUE THREAD

**Issue by d3vr4ndOm** [rswag/rswag#506]
Hello there,

unfortunately I was not able to write my "specs" without specifying the full `path`.
If I set the `servers: [ { url: 'localhost:3000' } ]` and specify the full `path '/api/v1/my_endpoint` in the spec, everything works as expected:
* bundle exec rspec spec/integration/* -> succeeds
* RAILS_ENV=test bundle exec rswag -> succeeds
* UI shows `Servers: http://localhost:3000` and endpoints are displayed as `/api/v1/my_endpoint` and a request -> succeeds

If I change the `servers: [ { url: 'localhost:3000' } ]` to `servers: [ url: { 'localhost:3000/api/v1' } ]` and change the `path '/api/v1/my_endpoint` to `path '/my_endpoint` I get:
* bundle exec rspec spec/integration* -> fails: `No route matches [*] "/my_endpoint"`
* RAILS_ENV=test bundle exec rswag -> succeeds
* UI shows `Servers: http://localhost:3000/api/v1` and endpoints are displayed as `/my_endpoint` and a request -> succeeds

My expectation would be that RSpec respects the full basepath, but maybe I am missing something?

```ruby
# spec/swagger_helper.rb
# frozen_string_literal: true

require 'rails_helper'

RSpec.configure do |config|
 config.swagger_root = Rails.root.join('swagger').to_s
 config.swagger_docs = {
   'v1/swagger.yaml' => {
      openapi: '3.0.1',
      info: {
        title: 'API V1',
        version: 'v1'
      },
      servers: [
        { url: 'http://localhost:3000/api/v1' }
      ]
    }
  }  
end
```

```ruby
# routes.rb

Rails.application.routes.draw do

  mount Rswag::Ui::Engine => '/api-docs'
  mount Rswag::Api::Engine => '/api-docs'

  namespace :api do
    namespace :v1, defaults: { format: 'json' } do
      resources :my_endpoint
    end
  end
end
```

## Relates to which version of OAS (Open API Specification)
- [ ] OAS2
- [x] OAS3
- [x] OAS3.1

**Comment by dollerbill** [comment]
~~@schlumpfit were you able to find any workarounds to this?~~

Looks like this was fixed in #547 and released in version 2.7.0 🙌

**Comment by d3vr4ndOm** [comment]
Thanks, @dollerbill for sharing the fix.
